gb/src/sdl_gfx_device.rs

```diff
@@ -7,33 +7,44 @@ pub struct SdlGfxDevice{
     _window_name: CString,
     renderer: *mut SDL_Renderer,
     texture: *mut SDL_Texture,
-    width:u32,
-    height:u32,
-    sacle:u8,
     discard:u8,
     turbo_mul:u8,
 }
 
 impl SdlGfxDevice{
-    pub fn new(buffer_width:u32, buffer_height:u32, window_name:&str, screen_scale: u8, turbo_mul:u8, disable_vsync:bool)->Self{
+    pub fn new(window_name:&str, screen_scale: u8, turbo_mul:u8, disable_vsync:bool, full_screen:bool)->Self{
         let cs_wnd_name = CString::new(window_name).unwrap();
 
         let (_window, renderer, texture): (*mut SDL_Window, *mut SDL_Renderer, *mut SDL_Texture) = unsafe{
             SDL_Init(SDL_INIT_VIDEO);
+
+            let window_flags = if full_screen{
+                SDL_WindowFlags::SDL_WINDOW_FULLSCREEN_DESKTOP as u32
+            }
+            else{
+                0
+            };
+
             let wind:*mut SDL_Window = SDL_CreateWindow(
                 cs_wnd_name.as_ptr(),
                 SDL_WINDOWPOS_UNDEFINED_MASK as i32, SDL_WINDOWPOS_UNDEFINED_MASK as i32,
-                buffer_width as i32 * screen_scale as i32, buffer_height as i32 * screen_scale as i32, 0);
-            let mut flags = SDL_RendererFlags::SDL_RENDERER_ACCELERATED as u32;
+                SCREEN_WIDTH as i32 * screen_scale as i32, SCREEN_HEIGHT as i32 * screen_scale as i32,
+                 window_flags);
+
+            let mut render_flags = SDL_RendererFlags::SDL_RENDERER_ACCELERATED as u32;
             if !disable_vsync{
-                flags |= SDL_RendererFlags::SDL_RENDERER_PRESENTVSYNC as u32;
+                render_flags |= SDL_RendererFlags::SDL_RENDERER_PRESENTVSYNC as u32;
             }
 
-            let rend: *mut SDL_Renderer = SDL_CreateRenderer(wind, -1, flags);
+            let rend: *mut SDL_Renderer = SDL_CreateRenderer(wind, -1, render_flags);
             
+            if SDL_RenderSetLogicalSize(rend, (SCREEN_WIDTH as u32) as i32, (SCREEN_HEIGHT as u32) as i32) != 0{
```

Comment:
Currently it decreases performance.

Check what happen if the texture is already in the correct size

Reply:
Aint no body got time for that